src/os_posix/os_thread.c

```diff
@@ -67,10 +67,32 @@ __wt_thread_join(WT_SESSION_IMPL *session, wt_thread_t tid)
 
 /*
  * __wt_thread_id --
+ *	Return an arithmetic representation of a thread ID on POSIX.
+ */
+void
+__wt_thread_id(size_t *id)
+    WT_GCC_FUNC_ATTRIBUTE((visibility("default")))
+{
+	pthread_t self;
+
+	/*
+	 * POSIX 1003.1 allows pthread_t to be an opaque type; on systems where
+	 * it's a pointer, print the pointer to match gdb output.
+	 */
+	self = pthread_self();
+#ifdef __sun
+	*id = (size_t)self;
+#else
+	*id = (size_t)(void *)self;
+#endif
```

Comment:
Are we sure this is going to work for all systems?  I recall the entire reason and history of why we ended up with `__wt_thread_id` as a string was because this sort of thing wasn't valid everywhere (maybe Mac?).  If you haven't I suggest looking into the historical ticket to make sure this is okay.